dll/shellext/netshell/lanstatusui.cpp

```diff
@@ -354,6 +354,45 @@ InsertItemToListView(
     return -1;
 }
 
+static BOOL tmToStr(
+    IN struct tm *pTM,
+    OUT WCHAR *szBuffer)
+{
+    SYSTEMTIME st;
+    LPWSTR szBufferDate;
+    LPWSTR szBufferTime;
+    UINT nCharDate, nCharTime;
+    BOOL bResult;
+
+    bResult = FALSE;
+    st.wYear = pTM->tm_year + 1900;
+    st.wMonth = pTM->tm_mon + 1;
+    st.wDay = pTM->tm_mday;
+    st.wHour = pTM->tm_hour;
+    st.wMinute = pTM->tm_min;
+    st.wSecond = pTM->tm_sec;            
+    
+    //Check required size before cpy/cat
+    nCharDate = GetDateFormatW(LOCALE_USER_DEFAULT, 0, &st, nullptr, szBufferDate, 0);
+    nCharTime = GetTimeFormatW(LOCALE_USER_DEFAULT, 0, &st, nullptr, szBufferDate, 0);
+    
+	szBufferDate = HeapAlloc(GetProcessHeap(), 0, nCharDate * sizeof(WCHAR));
+	szBufferTime = HeapAlloc(GetProcessHeap(), 0, nCharTime * sizeof(WCHAR));	
```

Comment:
Missing null checks. Indentation is off here, too (as well as on the `HeapFree`s below).

Reply:
added.